File: src/AudioManager.lua
AudioManager = AudioManager or {}

--[[
params:
1 name
2 dir
3 loop
]]--
function AudioManager.playEffect( ... )
    local args = { ... }
    local argc = #args
    local path = args[1]
    return true
--[[
    if argc > 1 then
        path = args[2] .. "/" .. path
    end
    path = "audio/" .. path
    local loop = argc > 2 and args[3] or false
    local fullpath = path .. ".wav"
    if not cc.FileUtils:getInstance():isFileExist( fullpath ) then
       fullpath = path .. ".mp3" 
    end
    return cc.SimpleAudioEngine:getInstance():playEffect( fullpath, loop )]]
end

function AudioManager.playMusic( ... )
    local args = { ... }
    local argc = #args
    local path = args[1]
    if argc > 1 then
        path = args[2] .. "/" .. path
    end
    path = "audio/" .. path
    local fullpath = path .. ".mp3"
    local loop = argc > 2 and args[3] or false
    if not cc.FileUtils:getInstance():isFileExist( fullpath ) then
        fullpath = path .. ".wav"
    end
    cc.SimpleAudioEngine:getInstance():playMusic( fullpath, loop )
end

function AudioManager.stopMusic()
    cc.SimpleAudioEngine:getInstance():stopMusic()
end 

File: frameworks/runtime-src/Classes/manager/AudioManager.cpp
//
//  AudioManager.cpp
//  Boids
//
//  Created by Yanjie Chen on 3/9/15.
//
//

#include "AudioManager.h"
#include "audio/include/SimpleAudioEngine.h"
#include "audio/include/AudioEngine.h"

using namespace cocos2d;
using namespace CocosDenshion;
using namespace experimental;

AudioManager* AudioManager::_instance = nullptr;

AudioManager::AudioManager() {

}

AudioManager::~AudioManager() {
    
}

AudioManager* AudioManager::getInstance() {
    if( _instance == nullptr ) {
        _instance = new AudioManager();
    }
    return _instance;
}

void AudioManager::destroy() {
    if( _instance ) {
        delete _instance;
        _instance = nullptr;
    }
}

void AudioManager::reset() {
    this->stopAll();
}

bool AudioManager::playMusic( const std::string& resource, bool loop ) {
    if( FileUtils::getInstance()->isFileExist( resource ) ) {
        SimpleAudioEngine::getInstance()->playBackgroundMusic( resource.c_str(), loop );
        return true;
    }
    
    return false;
}

bool AudioManager::playEffect( const std::string& resource, bool loop ) {
    if( FileUtils::getInstance()->isFileExist( resource ) ) {
        int aid = SimpleAudioEngine::getInstance()->playEffect( resource.c_str() );
        _audio_ids[resource] = Value( aid );
        return true;
    }
    return false;
}

void AudioManager::pauseMusic( const std::string& resource ) {
    SimpleAudioEngine::getInstance()->pauseBackgroundMusic();
}

void AudioManager::resumeMusic( const std::string& resource ) {
    SimpleAudioEngine::getInstance()->resumeBackgroundMusic();
}

void AudioManager::stopMusic( const std::string& resource ) {
    SimpleAudioEngine::getInstance()->stopBackgroundMusic();
}

void AudioManager::stopEffect( const std::string& resource ) {
    auto itr = _audio_ids.find( resource );
    if( itr != _audio_ids.end() ) {
        SimpleAudioEngine::getInstance()->stopEffect( itr->second.asInt() );
        _audio_ids.erase( itr );
    }
}

bool AudioManager::vibrate() {
    AudioEngine::vibrate();
    return true;
}

void AudioManager::stopAll() {
    SimpleAudioEngine::getInstance()->stopAllEffects();
    SimpleAudioEngine::getInstance()->stopBackgroundMusic();
    AudioEngine::stopAll();
    _audio_ids.clear();
}